contracts/chain-adapters/Arbitrum_Adapter.sol

```diff
@@ -1,78 +1,59 @@
-    // includes is sending messages from L1 to L2.
-    uint256 public constant L2_CALL_VALUE = 0;
-
-    // Gas limit for L2 execution of a cross chain token transfer sent via the inbox.
-    uint32 public constant RELAY_TOKENS_L2_GAS_LIMIT = 300_000;
-    // Gas limit for L2 execution of a message sent via the inbox.
-    uint32 public constant RELAY_MESSAGE_L2_GAS_LIMIT = 2_000_000;
-
-    address public constant L1_DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
-
-    // This address on L2 receives extra ETH that is left over after relaying a message via the inbox.
-    address public immutable L2_REFUND_L2_ADDRESS;
-
-    // Inbox system contract to send messages to Arbitrum. Token bridges use this to send tokens to L2.
-    // https://github.com/OffchainLabs/nitro-contracts/blob/f7894d3a6d4035ba60f51a7f1334f0f2d4f02dce/src/bridge/Inbox.sol
-    ArbitrumL1InboxLike public immutable L1_INBOX;
-
-    // Router contract to send tokens to Arbitrum. Routes to correct gateway to bridge tokens. Internally this
-    // contract calls the Inbox.
-    // Generic gateway: https://github.com/OffchainLabs/token-bridge-contracts/blob/main/contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol
-    ArbitrumL1ERC20GatewayLike public immutable L1_ERC20_GATEWAY_ROUTER;
-
     /**
      * @notice Constructs new Adapter.
      * @param _l1ArbitrumInbox Inbox helper contract to send messages to Arbitrum.
      * @param _l1ERC20GatewayRouter ERC20 gateway router contract to send tokens to Arbitrum.
      * @param _l2RefundL2Address L2 address to receive gas refunds on after a message is relayed.
      * @param _l1Usdc USDC address on L1.
      * @param _cctpTokenMessenger TokenMessenger contract to bridge via CCTP.
+     * @param _circleDomainId Circle CCTP domain ID for the target network (3 for Arbitrum).
+     * @param _l2MaxSubmissionCost maximum amount of ETH to send with a transaction for it to execute on L2.
+     * @param _l2GasPrice gas price bid for a message to be executed on L2.
      */
     constructor(
-        ArbitrumL1InboxLike _l1ArbitrumInbox,
+        ArbitrumInboxLike _l1ArbitrumInbox,
         ArbitrumL1ERC20GatewayLike _l1ERC20GatewayRouter,
         address _l2RefundL2Address,
         IERC20 _l1Usdc,
-        ITokenMessenger _cctpTokenMessenger
-    ) CircleCCTPAdapter(_l1Usdc, _cctpTokenMessenger, CircleDomainIds.Arbitrum) {
-        L1_INBOX = _l1ArbitrumInbox;
-        L1_ERC20_GATEWAY_ROUTER = _l1ERC20GatewayRouter;
-        L2_REFUND_L2_ADDRESS = _l2RefundL2Address;
-    }
+        ITokenMessenger _cctpTokenMessenger,
+        uint32 _circleDomainId,
+        uint256 _l2MaxSubmissionCost,
+        uint256 _l2GasPrice
+    )
+        Arbitrum_AdapterBase(
+            _l1ArbitrumInbox,
+            _l1ERC20GatewayRouter,
+            _l2RefundL2Address,
+            _l1Usdc,
+            _cctpTokenMessenger,
+            _circleDomainId,
+            _l2MaxSubmissionCost,
```

Comment:
Note for future work: anytime you make changes to existing contract code where its understood that you're refactoring, you should leave a comment saying something like "I changed this code here"